aidb/estimator/estimator.py

```diff
@@ -59,16 +58,15 @@ def __init__(self, population_size: int) -> None:
 
 
   @abc.abstractmethod
-  def estimate(self, samples: List[SampledBlob], num_samples: int, conf: float, **kwargs) -> Estimate:
+  def estimate(self, samples: List[SampledBlob], num_samples: int, conf: float, agg_index: int, **kwargs) -> Estimate:
```

Comment:
why agg index? seems very weird to pass in an index to a function like this...

Reply:
https://github.com/ddkang/aidb/blob/76ac2a68d414ed434a83e5f56c5e9e8b9cf0ae53/aidb/engine/approx_aggregate_engine.py#L82-L95
https://github.com/ddkang/aidb/blob/76ac2a68d414ed434a83e5f56c5e9e8b9cf0ae53/aidb/samplers/sampler.py#L18-L25
In estimator function, I retrieve the result externally rather than passing an index. However, I'm concerned this might lead to increased memory usage. Is this an issue?

Reply:
I don't think I understand the problem, can you describe it in more detail?

Reply:
https://github.com/ddkang/aidb/blob/76ac2a68d414ed434a83e5f56c5e9e8b9cf0ae53/aidb/samplers/sampler.py#L8-L15
The result has such a data structure. If we want to avoid passing an index. We need to extract statistic with weight, mass and num_items for each index first. And then we pass the extracted result to estimator to estimate. Is it fine?

Reply:
This might not be the best way to have a SampledBlob. It may even be better to turn it into a flat dataframe that is pass along whole. Don't take design decisions made in the past as perfect